File: Predict the Column/Predict the Column.java
class Solution
{
    int columnWithMaxZeros(int arr[][], int N)
    {
        // code here 
        int max = Integer.MIN_VALUE;
        int index = -1;
        for(int i = 0; i < arr.length; i++) {
            int count = 0;
            for (int j = 0; j < arr.length; j++) {
                if(arr[j][i] == 0) {
                    count++;
              }
            }
            if(count == arr.length) {
                return i;
            }
            
            if(count > max && count != 0) {
                index = i;
                max = count;
            }
        
        
    }
    
    return index;
    }
}

File: Predict the Column/Predict the Column.cpp

class Solution{
    public:
    /*Function to count zeros in each column
    * N : Number of rows and columns in array
    M is the matrix that is globally declared
    */
    int columnWithMaxZeros(vector<vector<int>>arr,int N){
        
        // Your code here
                    int max = INT_MIN;
              int index = -1;
            
              for (int i = 0; i < arr.size(); ++i) {
                int count = 0;
            
                for (int j = 0; j < arr.size(); ++j) {
                  if (arr[j][i] == 0) {
                    count++;
                  }
                }
            
                if (count == arr.size()) {
                  return i;
                }
            
                if (count > max && count != 0) {
                  index = i;
                  max = count;
                }
              }
            
              return index;
                    
        
    }
};
